frontend/src/services/ApiService.ts

```diff
@@ -19,22 +19,27 @@ export class HttpError extends Error {
  * @returns HttpError if AxiosError, else original error
  */
 export const transformAxiosError = (e: Error): ApiError => {
-  if (axios.isAxiosError(e) && e.response) {
-    const statusCode = e.response.status
-    if (statusCode === 429) {
-      return new HttpError('Please try again later.', statusCode)
-    }
-    if (typeof e.response.data === 'string') {
-      return new HttpError(e.response.data, statusCode)
-    }
-    if (e.response.data?.message) {
-      return new HttpError(e.response.data.message, statusCode)
-    }
-    if (e.response.statusText) {
-      return new HttpError(e.response.statusText, statusCode)
-    }
-
-    return new HttpError(`Http ${statusCode} error`, statusCode)
+  if (axios.isAxiosError(e)) {
+    if (e.response) {
+      const statusCode = e.response.status
+      if (statusCode === 429) {
+        return new HttpError('Please try again later.', statusCode)
+      }
+      if (typeof e.response.data === 'string') {
+        return new HttpError(e.response.data, statusCode)
+      }
+      if (e.response.data?.message) {
+        return new HttpError(e.response.data.message, statusCode)
+      }
+      if (e.response.statusText) {
+        return new HttpError(e.response.statusText, statusCode)
+      }
+      return new HttpError(`Http ${statusCode} error`, statusCode)
+    } else if (e.request) {
+      return new Error(
+        'There was a problem with your internet connection. Please check your network and try again.',
+      )
+    } else return e
```

Comment:
This `else` is not needed since line 44 does it already?

Reply:
its actually needed, since the `ifs` are now nested. if it's an axios error that has no request or response, we don't return anything in that case

Reply:
SORRY LOL i brainfarted, will make this change.